DataStructures/BitList.cs

```diff
@@ -0,0 +1,1753 @@
+
+namespace DataStructures
+{
+    /// <summary>
+    /// Exception generated calling a method when the BitList instance is empty.
+    /// </summary>
+    public class EmptyInstanceException: Exception { }
+
+    /// <summary>
+    /// Exception generated when a boolean operation fails.
+    /// </summary>
+    public class BooleanAlgebraException: Exception { }
+
+    /// <summary>
+    /// Exception generated when a BitList instance fails to write or read a file.
+    /// </summary>
+    public class FileException: Exception { }
+
+    /// <summary>
+    /// Exception generated when a BitList instance fails to read a file.
+    /// </summary>
+    public class FileReadingException: FileException { }
+
+    /// <summary>
+    /// Exception generated when a BitList instance fails to write a file.
+    /// </summary>
+    public class FileWritingException: FileException { }
+
+    /// <summary>
+    /// Exception generated when a BitList instance fails the conversion to another type.
+    /// </summary>
+    public class ConversionErrorException: Exception { }
+
+    /// <summary>
+    /// A list of bits.
+    /// </summary>
+    /// <exception cref="EmptyInstanceException" />
+    /// <exception cref="BooleanAlgebraException" />
+    /// <exception cref="FileReadingException" />
+    /// <exception cref="FileWritingException" />
+    /// <exception cref="ConversionErrorException" />
+    public sealed class BitList:
+        IList<bool>,
+        IEquatable<BitList>
+    {
+        private BitArray bits;
+
+        /// <summary>
+        /// No params ctor.
+        /// </summary>
+        public BitList()
+        {
+            bits = null;
+        }
+
+        /// <summary>
+        /// Ctor (Copy the <paramref name="bits"/> <see cref="BitList"/> object).
+        /// </summary>
+        public BitList(BitList bits)
+        {
+            this.bits = new BitArray(bits.bits);
+        }
+
+        /// <summary>
+        /// Ctor (<paramref name="unmanagedNonPrimitiveBits"/> is an unmanaged value type object).
+        /// </summary>
+        public BitList(ValueType unmanagedNonPrimitiveBits)
+        {
+            //try
+            //{
+            if (unmanagedNonPrimitiveBits.GetType().IsPrimitive)
+            {
+                throw new Exception();
+            }
+
+            //Throw an exception if the type of unmanagedBits isn't unmanaged; otherwise, continue.
+            typeof(Unmanaged<>).MakeGenericType(unmanagedNonPrimitiveBits.GetType());
+
+            //Unit of measure: byte (8 bit).
+            var size = Marshal.SizeOf(
+                unmanagedNonPrimitiveBits.GetType().IsEnum ?
+                    Enum.GetUnderlyingType(unmanagedNonPrimitiveBits.GetType()) : unmanagedNonPrimitiveBits.GetType()
+            );
+
+            if (unmanagedNonPrimitiveBits.GetType().IsEnum)
+            {
+                switch (Enum.GetUnderlyingType(unmanagedNonPrimitiveBits.GetType()).Name)
```

Comment:
Codacy found an issue: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)